applications/FluidDynamicsApplication/custom_utilities/mass_conservation_utility.cpp

```diff
@@ -169,46 +248,38 @@ void MassConservationCheckProcess::ComputeVolumesAndInterface( double& positiveV
         Matrix shape_functions;
         GeometryType::ShapeFunctionsGradientsType shape_derivatives;
 
-        const auto rGeom = it_elem->GetGeometry();
+        auto r_geometry = it_elem->GetGeometry();
         unsigned int pt_count_pos = 0;
         unsigned int pt_count_neg = 0;
+        const bool is_geometry_cut = IsGeometryCut(r_geometry, pt_count_neg, pt_count_pos);
 
-        // instead of using data.isCut()
-        for (unsigned int pt = 0; pt < rGeom.Points().size(); pt++){
-            if ( rGeom[pt].FastGetSolutionStepValue(DISTANCE) > 0.0 ){
-                pt_count_pos++;
-            } else {
-                pt_count_neg++;
-            }
-        }
-
-        if ( pt_count_pos == rGeom.PointsNumber() ){
+        if ( pt_count_pos == r_geometry.PointsNumber() ){
             // all nodes are positive (pointer is necessary to maintain polymorphism of DomainSize())
             pos_vol += it_elem->pGetGeometry()->DomainSize();
         }
-        else if ( pt_count_neg == rGeom.PointsNumber() ){
+        else if ( pt_count_neg == r_geometry.PointsNumber() ){
             // all nodes are negative (pointer is necessary to maintain polymorphism of DomainSize())
             neg_vol += it_elem->pGetGeometry()->DomainSize();
         }
-        else if ( 0 < pt_count_neg && 0 < pt_count_pos ){
+        else if ( is_geometry_cut ){
             // element is cut by the surface (splitting)
-            Kratos::unique_ptr<ModifiedShapeFunctions> p_modified_sh_func = nullptr;
             Vector w_gauss_pos_side(3, 0.0);
             Vector w_gauss_neg_side(3, 0.0);
             Vector w_gauss_interface(3, 0.0);
 
-            Vector Distance( rGeom.PointsNumber(), 0.0 );
-            for (unsigned int i = 0; i < rGeom.PointsNumber(); i++){
+            Vector nodal_distances( r_geometry.PointsNumber(), 0.0 );
+            for (unsigned int i = 0; i < r_geometry.PointsNumber(); i++){
                 // Control mechanism to avoid 0.0 ( is necessary because "distance_modification" possibly not yet executed )
-                if ( rGeom[i].FastGetSolutionStepValue(DISTANCE) == 0.0 ){
-                    it_elem->GetGeometry().GetPoint(i).FastGetSolutionStepValue(DISTANCE) = 1.0e-7;
+                double &r_dist = r_geometry[i].FastGetSolutionStepValue(DISTANCE);
```

Comment:
Right now we always assume that the level set is always stored in the `DISTANCE` variable. Although this is the case in the vast majority of cases, it might not be always the case.

Maybe makes sense that we take the chance to add a `const Variable<double>& mrLevelSetVariable` member variable, which default value is `DISTANCE`.